### ISSUE TITLE
Reproducibility with versioning

### ISSUE THREAD

**Issue by lahr-ul** [tensorflow/datasets#3068]
I created a custom dataset with `tfds build` with a single train split. I want to share this dataset with my colleagues and ensure reproducibility such that if we all use a `80:10:10` split on the dataset, the split sets yield the same data on all our workstations. The [versioning section](https://www.tensorflow.org/datasets/datasets_versioning#semantic) in the documentation states that as long as the major version remains unchanged between runs, "the data as read by the client is the same". Does that mean that me and my colleagues have the same contents in the splits if everyone uses version `1.0.0`?

**Comment by Conchylicultor** [comment]
Yes, examples are deterministically shuffled sorted using the hash of the `key`, yield by `_generate_examples`. 

If you don't observe this, it means there is likely a bug in the dataset implementation which does not yield deterministic key. See https://www.tensorflow.org/datasets/api_docs/python/tfds/core/GeneratorBasedBuilder#_generate_examples for details. The key should not contain any user system specific (e.g. key should not be an absolute path). If using `enumerate()` for the key, make sure that examples are always generated in the same order.

You can check by comparing the first `tfds.load(..., split=['80%:']).take(3)` with your coworker.
